### ISSUE TITLE
Oni2 crashes when opening *.rst file

### ISSUE THREAD

**Issue by gour** [onivim/oni2#1471]
Hello,

I've the following extensions installed:
```
$ oni2 --list-extensions                                                               
python
restructuredtext
```

and after opening *.rst file, Oni2 crashes with the following log:

[oni2-rst.log](https://github.com/onivim/oni2/files/4332520/oni2-rst.log)

**Comment by glennsl** [comment]
Thanks for the report @gour. From the log I can see that the extension loaded successfully, and I can't see anything else suspicious. Could you try again with `--trace` to get more verbose logging?

Also, is this specific to one `.rst` file, or doers it happen with any?

Is there an `onivim2-crash.log` in the directory you ran it from? If we're able to catch the exception we log the exception information there.

**Comment by gour** [comment]
> Is there an onivim2-crash.log in the directory you ran it from? If we're able to catch the exception we log the exception information there.

```
Yojson.Safe.Util.Type_error("Expected array, got null", 870828711):
```

**Comment by glennsl** [comment]
Oh, that's promising. Seems like it is sending us something we don't expect. Could you try run it again with `OCAMLRUNPARAM=b` set? That should capture the stack trace as well.

```
OCAMLRUNPARAM=b oni2 -f --trace`
```

will log the stack trace and give more detailed logging of what's being sent back and forth.

**Comment by gour** [comment]
> Could you try again with --trace to get more verbose logging?
In the meantime I've installed few more extensions....
```
$ oni2 --list-extensions                                                               
vscode-tailwindcss
context-syntax
fossil
pinegrow-vscode
jinjahtml
jinja
html-css-class-completion
restructuredtext
```
> Also, is this specific to one .rst file, or doers it happen with any?

...and it crashes with other 1.1K ConTeXt  file as well. 

Here is the new crash-log:

```
Yojson.Safe.Util.Type_error("Expected array, got null", 870828711):
Raised at file "util.ml", line 15, characters 20-60
Called from file "util.ml" (inlined), line 19, characters 13-19
Called from file "src/Extensions/ExtHostClient_SCM.re", line 165, characters 31-68
Called from file "util.ml" (inlined), line 19, characters 13-19
Called from file "src/Extensions/ExtHostClient_SCM.re", line 157, characters 14-427
Called from file "src/Extensions/ExtHostClient.re", line 224, characters 6-74
Called from file "src/Extensions/ExtHostTransport.re", line 128, characters 13-53
Called from file "list.ml", line 110, characters 12-15
Called from file "src/Core/App.re", line 237, characters 6-100
Called from file "src/sdl2.re", line 648, characters 10-20
Called from file "src/bin_editor/Oni2_editor.re", line 236, characters 2-18
```
and here is the trace.log:

[oni2-trace.log](https://github.com/onivim/oni2/files/4332714/oni2-trace.log)

**Comment by gour** [comment]
> Could you try run it again with OCAMLRUNPARAM=b set?

Here is the log:

[oni2-trace-b.log](https://github.com/onivim/oni2/files/4332879/oni2-trace-b.log)

**Comment by gour** [comment]
And here is the crash.log:

```
Yojson.Json_error("Line 1, bytes 1-34:\nExpected string or identifier but found '?xml version=\"1.0\" encoding=\"UTF-'"):
Raised at file "common.ml", line 5, characters 19-39
Called from file "lib/read.mll", line 290, characters 32-51
Called from file "lib/read.ml" (inlined), line 1025, characters 3-39
Called from file "lib/read.mll", line 1096, characters 8-26
Called from file "lib/read.mll", line 1117, characters 6-26
Called from file "lib/read.mll", line 1124, characters 14-47
Re-raised at file "lib/read.mll", line 1129, characters 6-13
Called from file "src/Syntax/GrammarRepository.re", line 28, characters 17-51
Called from file "src/GrammarRepository.re" (inlined), line 8, characters 40-57
Called from file "src/Tokenizer.re", line 22, characters 9-60
Called from file "src/Syntax/TextmateTokenizerJob.re", line 120, characters 6-204
Called from file "src/Core/Job.re", line 69, characters 4-33
Called from file "src/Core/Job.re" (inlined), line 73, characters 31-42
Called from file "src/Core/Job.re", line 96, characters 15-31
Called from file "src/Feature/Syntax/Feature_Syntax.re", line 16, characters 4-151
Called from file "src/Store/SyntaxHighlightingStoreConnector.re", line 76, characters 8-152
Called from file "src/Updater.re", line 14, characters 33-56
Called from file "list.ml", line 121, characters 24-34
Called from file "src/Updater.re", line 11, characters 4-253
Called from file "src/Store.re", line 113, characters 29-61
Called from file "list.ml", line 110, characters 12-15
Called from file "list.ml", line 110, characters 12-15
Called from file "src/Buffer.re", line 8, characters 2-46
Called from file "src/Store/VimStoreConnector.re", line 623, characters 19-48
Called from file "list.ml", line 110, characters 12-15
Called from file "src/Effect.re" (inlined), line 31, characters 20-38
Called from file "src/Store.re", line 174, characters 9-32
Called from file "list.ml", line 110, characters 12-15
Called from file "src/Core/Tick.re", line 112, characters 9-15
Called from file "src/Core/Tick.re", line 90, characters 15-32
Called from file "list.ml", line 92, characters 20-23
Called from file "list.ml", line 92, characters 32-39
Called from file "src/Core/Tick.re", line 99, characters 22-52
Called from file "src/Core/App.re", line 227, characters 4-23
Called from file "src/sdl2.re", line 648, characters 10-20
Called from file "src/bin_editor/Oni2_editor.re", line 236, characters 2-18
```

**Comment by glennsl** [comment]
Thanks! The last one is a known issue. We don't support XML grammars and themes yet. See #912, which has a workaround if you really need it.

The other crash I've submitted a fix for, in #1472. Once that's merged, it should be included in the nightly build tomorrow. Or if you're feeling adventurous you could try building from source.

**Comment by gour** [comment]
And here is the crash.log:

```
Yojson.Json_error("Line 1, bytes 1-34:\nExpected string or identifier but found '?xml version=\"1.0\" encoding=\"UTF-'"):
Raised at file "common.ml", line 5, characters 19-39
Called from file "lib/read.mll", line 290, characters 32-51
Called from file "lib/read.ml" (inlined), line 1025, characters 3-39
Called from file "lib/read.mll", line 1096, characters 8-26
Called from file "lib/read.mll", line 1117, characters 6-26
Called from file "lib/read.mll", line 1124, characters 14-47
Re-raised at file "lib/read.mll", line 1129, characters 6-13
Called from file "src/Syntax/GrammarRepository.re", line 28, characters 17-51
Called from file "src/GrammarRepository.re" (inlined), line 8, characters 40-57
Called from file "src/Tokenizer.re", line 22, characters 9-60
Called from file "src/Syntax/TextmateTokenizerJob.re", line 120, characters 6-204
Called from file "src/Core/Job.re", line 69, characters 4-33
Called from file "src/Core/Job.re" (inlined), line 73, characters 31-42
Called from file "src/Core/Job.re", line 96, characters 15-31
Called from file "src/Feature/Syntax/Feature_Syntax.re", line 16, characters 4-151
Called from file "src/Store/SyntaxHighlightingStoreConnector.re", line 76, characters 8-152
Called from file "src/Updater.re", line 14, characters 33-56
Called from file "list.ml", line 121, characters 24-34
Called from file "src/Updater.re", line 11, characters 4-253
Called from file "src/Store.re", line 113, characters 29-61
Called from file "list.ml", line 110, characters 12-15
Called from file "list.ml", line 110, characters 12-15
Called from file "src/Buffer.re", line 8, characters 2-46
Called from file "src/Store/VimStoreConnector.re", line 623, characters 19-48
Called from file "list.ml", line 110, characters 12-15
Called from file "src/Effect.re" (inlined), line 31, characters 20-38
Called from file "src/Store.re", line 174, characters 9-32
Called from file "list.ml", line 110, characters 12-15
Called from file "src/Core/Tick.re", line 112, characters 9-15
Called from file "src/Core/Tick.re", line 90, characters 15-32
Called from file "list.ml", line 92, characters 20-23
Called from file "list.ml", line 92, characters 32-39
Called from file "src/Core/Tick.re", line 99, characters 22-52
Called from file "src/Core/App.re", line 227, characters 4-23
Called from file "src/sdl2.re", line 648, characters 10-20
Called from file "src/bin_editor/Oni2_editor.re", line 236, characters 2-18
```

**Comment by gour** [comment]
> The other crash I've submitted a fix for, in #1472. 

But, originally, I didn't have Fossil extension installed?

**Comment by glennsl** [comment]
There could of course be more bugs lurking. If you remove the fossil extension and fix or avoid opening rst files, we can find out if it still crashes then. And if it does, gather information the same way.

**Comment by glennsl** [comment]
Should have been fixed #1616 and #1649, which picked up https://github.com/onivim/reason-textmate/pull/57. Let us know if there's still problems though!
